README.md
# TYPO3 Extension core_upgrader

[![Latest Stable Version](https://img.shields.io/packagist/v/ichhabrecht/core-upgrader.svg)](https://packagist.org/packages/ichhabrecht/core-upgrader)
[![Build Status](https://img.shields.io/travis/IchHabRecht/core_upgrader/master.svg)](https://travis-ci.org/IchHabRecht/core_upgrader)
[![StyleCI](https://styleci.io/repos/263364343/shield?branch=master)](https://styleci.io/repos/263364343)

Run upgrade wizards for multiple TYPO3 versions (to 10.4) at once.

## Features

This extension allows to upgrade the TYPO3 core from v7.6 to v10.4 in one step.

## Installation

Simply install the extension with Composer or download from [TER](https://extensions.typo3.org/extension/core_upgrader/).

`composer require ichhabrecht/core-upgrader`

This extension depends on [TYPO3 Console](https://github.com/TYPO3-Console/TYPO3-Console).
The typo3cms binary will be installed in the specified bin-dir (by default `vendor/bin`).

## Usage

1. You need to activate the extension:

    `typo3cms install:generatepackagestates`

2. Now you can run all update wizards:

     `typo3cms coreupgrader:upgrade`

 3. The upgrade command runs necessary TYPO3 upgrade wizards.\
    It is recommended to run TYPO3 Console upgrade command afterwards to execute confirmable and extension wizards. 
    
     `typo3cms upgrade:run all`

## Community

- Thanks to [elementare teilchen GmbH](https://www.elementare-teilchen.de) that sponsors the maintenance of this extension


ChangeLog
2021-02-21         [RELEASE] Release of core_upgrader 1.3.1 (Nicole Cordes)
2021-02-19 bca9706 [BUGFIX] Prevent errors in EXT:scheduler backend module (Nicole Cordes)
2020-10-25 1ac3de5 [TASK] Update supporting files to EXT:my_skeleton 0.5.0 (Nicole Cordes)

2020-09-08         [RELEASE] Release of core_upgrader 1.3.0 (Nicole Cordes)
2020-09-04 46bbe55 [FEATURE] Add additional EXT:form upgrade wizard (Nicole Cordes)
2020-09-04 9fe865c [BUGFIX] Ensure additional upgrade wizards exists (Nicole Cordes)

2020-08-17         [RELEASE] Release of core_upgrader 1.2.1 (Nicole Cordes)
2020-08-17 c8ae286 Use TYPO3 RowUpdaterInterface (Nicole Cordes)
2020-08-17 6d5ccae Streamline wizard identifier (Nicole Cordes)
2020-08-17 491b93d Remove ext_emconf.php changes (Nicole Cordes)
2020-08-11 4a165f4 [BUGFIX] DatabaseRowsUpdateWizard of TYPO3 V8 fixed (Mohsin Khan)

2020-08-05         [RELEASE] Release of core_upgrader 1.2.0 (Nicole Cordes)
2020-08-05 82b356f [TASK] Update supporting files to EXT:my_skeleton 0.4.0 (Nicole Cordes)
2020-08-03 029f0ce [FEATURE] Add missing TYPO3\CMS updates (Nicole Cordes)

2020-05-26         [RELEASE] Release of core_upgrader 1.1.0 (Nicole Cordes)
2020-05-26 bbc2a33 [TASK] Remove dev-master tests from Travis CI configuration (Nicole Cordes)
2020-05-26 ce29717 [FEATURE] Allow to force update wizards (Nicole Cordes)

2020-05-19         [RELEASE] Release of core_upgrader 1.0.0 (Nicole Cordes)
2020-05-15 9bdc23b [TASK] Recommend to run all upgrade wizards (Nicole Cordes)
2020-05-15 511eeeb [FEATURE] Add upgrade wizards for TYPO3 10.4 (Nicole Cordes)
2020-05-15 dc72a93 [FEATURE] Add upgrade wizards for TYPO3 9.5 (Nicole Cordes)
2020-05-15 86eeed9 [FEATURE] Add upgrade wizards for TYPO3 8.7 (Nicole Cordes)
2020-05-15 636513d [FEATURE] Introduce upgrade command (Nicole Cordes)
2020-05-15 8edb518 [TASK] Add basic extension files (Nicole Cordes)


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "core_upgrader".
 *
 * Auto generated 21-02-2021 18:57
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Core upgrader',
  'description' => 'Run upgrade wizards for multiple TYPO3 versions at once',
  'category' => 'cli',
  'author' => 'Nicole Cordes',
  'author_email' => 'typo3@cordes.co',
  'author_company' => 'biz-design',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearcacheonload' => 0,
  'version' => '1.3.1',
  'constraints' => 
  array (
    'depends' => 
    array (
      'typo3' => '10.4.0-10.4.99',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:31:{s:9:"ChangeLog";s:4:"f707";s:7:"LICENSE";s:4:"b234";s:9:"README.md";s:4:"642f";s:13:"composer.json";s:4:"b3c8";s:12:"ext_icon.png";s:4:"0ef7";s:17:"ext_localconf.php";s:4:"4bc5";s:16:"phpunit.xml.dist";s:4:"041c";s:24:"sonar-project.properties";s:4:"4769";s:34:"Classes/Command/UpgradeCommand.php";s:4:"4397";s:50:"Classes/Updates/v87/BulletContentElementUpdate.php";s:4:"bb9a";s:59:"Classes/Updates/v87/CommandLineBackendUserRemovalUpdate.php";s:4:"07b5";s:54:"Classes/Updates/v87/Compatibility7ExtractionUpdate.php";s:4:"bfba";s:48:"Classes/Updates/v87/DatabaseRowsUpdateWizard.php";s:4:"ffc0";s:43:"Classes/Updates/v87/FileReferenceUpdate.php";s:4:"c7c1";s:50:"Classes/Updates/v87/FillTranslationSourceField.php";s:4:"e8be";s:50:"Classes/Updates/v87/FormLegacyExtractionUpdate.php";s:4:"1971";s:53:"Classes/Updates/v87/FrontendUserImageUpdateWizard.php";s:4:"f15a";s:45:"Classes/Updates/v87/LanguageSortingUpdate.php";s:4:"8d07";s:50:"Classes/Updates/v87/MigrateFeSessionDataUpdate.php";s:4:"6db7";s:54:"Classes/Updates/v87/MigrateFscStaticTemplateUpdate.php";s:4:"6bf2";s:51:"Classes/Updates/v87/RteHtmlAreaExtractionUpdate.php";s:4:"11bf";s:54:"Classes/Updates/v87/SectionFrameToFrameClassUpdate.php";s:4:"8c83";s:40:"Classes/Updates/v87/SplitMenusUpdate.php";s:4:"6633";s:50:"Classes/Updates/v87/UploadContentElementUpdate.php";s:4:"a5bc";s:44:"Classes/Updates/v87/WizardDoneToRegistry.php";s:4:"3d14";s:51:"Classes/Updates/v87/RowUpdater/ImageCropUpdater.php";s:4:"b3b3";s:50:"Classes/Updates/v87/RowUpdater/L10nModeUpdater.php";s:4:"ff27";s:55:"Classes/Updates/v87/RowUpdater/RteLinkSyntaxUpdater.php";s:4:"71f6";s:26:"Configuration/Commands.php";s:4:"7ee1";s:26:"Configuration/Upgrades.php";s:4:"1a41";s:36:"Resources/Public/Icons/Extension.svg";s:4:"960d";}',
);

